numpy/ma/core.py

```diff
@@ -7345,6 +7345,48 @@ def putmask(a, mask, values):  # , mode='raise'):
     return
 
 
+def matrix_transpose(a):
+    """
+    View of the matrix transposed array.
+
+    The matrix transpose is the transpose of the last two dimensions, even
+    if the array is of higher dimension.
+
+    .. versionadded:: 2.0
+
+    Raises
+    ------
+    ValueError
+        If the array is of dimension less than 2.
+
+    Examples
+    --------
+    >>> import numpy.ma as ma
+    >>> x = ma.arange(4).reshape((2,2))
+    >>> x[1, 1] = ma.masked
+    >>> x
+    masked_array(
+      data=[[0, 1],
+            [2, --]],
+      mask=[[False, False],
+            [False,  True]],
+      fill_value=999999)
+
+    >>> ma.matrix_transpose(x)
+    masked_array(
+      data=[[0, 2],
+            [1, --]],
+      mask=[[False, False],
+            [False,  True]],
+      fill_value=999999)
+
+    """
+    try:
+        return a.mT
+    except AttributeError:
+        return narray(a, copy=False).mT.view(MaskedArray)
```

Comment:
Should we skip the `a.mT`, since you didn't do that in the array path either?  I dunno, this function is different from others with a method/attribute (since NumPy historically tries to use those aggressively).

This one question, otherwise, LGTM.

Reply:
I'm happy to skip it. Realistically, I think the overhead of a try/except will be very small / negligible (although I haven't tested it). The reason I did it this was that I was copying `np.ma.transpose`

Reply:
I actually think we can use `_frommethod` here. `np.ma.MaskedArray.mT` doesn't actually require any arguments so it should be pretty simple, let me try that

Reply:
I was only thinking about it because we also don't fall back to method/property for arrays which we do for `transpose`.  But it's OK let's just leave it as is, at least the normal function has `__array_function__`.  And if someone comes and wants to change it, we can.

Reply:
I have one more question regarding this: at the moment, `np.ma.matrix_transpose` will return a regular array if the input is a regular array, do we want that?

Reply:
I don't really like it, and you effectively changed something like that in `matrix_transpose` compared to `transpose`. But... I am OK with just doing this, because I think masked arrays need a bigger cleanup, and I don't know if this is somehow important elsewhere.